framework/src/outputs/XMLOutput.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Moose includes
+#include "XMLOutput.h"
+#include "FEProblem.h"
+#include "MooseApp.h"
+
+registerMooseObject("MooseApp", XMLOutput);
+
+defineLegacyParams(XMLOutput);
```

Comment:
When is it that we actually need to define legacy parameters?

Reply:
1. We need to update the framework and all the apps. 
2. Then add a legacy flag to determine which function to call
3. Update stork to only have the new ones
4. Update all the apps to use the new ones
5. Remove old